roachtest: gorm failed

roachtest.gorm [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6837141&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6837141&tab=artifacts#/gorm) on release-22.1 @ [0ae8b84678937188c66180005a3099e552a4b896](https://github.com/cockroachdb/cockroach/commits/0ae8b84678937188c66180005a3099e552a4b896):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/gorm/run_1
	orm_helpers.go:193,orm_helpers.go:119,java_helpers.go:220,gorm.go:127,test_runner.go:883: 
		Tests run on Cockroach v22.1.8-104-g0ae8b84678
		Tests run against gorm v1.24.0
		1 Total Tests Run
		0 tests passed
		1 test failed
		0 tests skipped
		0 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.[build failed] - unknown (unexpected)
		For a full summary look at the gorm artifacts 
		
		An updated blocklist (gormBlocklist22_1) is available in the artifacts' gorm log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*gorm.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-20328

roachtest.gorm [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6848265&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6848265&tab=artifacts#/gorm) on release-22.1 @ [0ae8b84678937188c66180005a3099e552a4b896](https://github.com/cockroachdb/cockroach/commits/0ae8b84678937188c66180005a3099e552a4b896):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/gorm/run_1
	orm_helpers.go:193,orm_helpers.go:119,java_helpers.go:220,gorm.go:127,test_runner.go:883: 
		Tests run on Cockroach v22.1.8-104-g0ae8b84678
		Tests run against gorm v1.24.0
		1 Total Tests Run
		0 tests passed
		1 test failed
		0 tests skipped
		0 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.[build failed] - unknown (unexpected)
		For a full summary look at the gorm artifacts 
		
		An updated blocklist (gormBlocklist22_1) is available in the artifacts' gorm log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89606 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2 release-blocker]
- #89605 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-master release-blocker]
- #89604 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2.0 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*gorm.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.gorm [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6859841&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6859841&tab=artifacts#/gorm) on release-22.1 @ [8c89608e52d71f4cb369b1fea2f407186637c19f](https://github.com/cockroachdb/cockroach/commits/8c89608e52d71f4cb369b1fea2f407186637c19f):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/gorm/run_1
	orm_helpers.go:193,orm_helpers.go:119,java_helpers.go:220,gorm.go:127,test_runner.go:883: 
		Tests run on Cockroach v22.1.8-106-g8c89608e52
		Tests run against gorm v1.24.0
		1 Total Tests Run
		0 tests passed
		1 test failed
		0 tests skipped
		0 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.[build failed] - unknown (unexpected)
		For a full summary look at the gorm artifacts 
		
		An updated blocklist (gormBlocklist22_1) is available in the artifacts' gorm log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89606 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2 release-blocker]
- #89605 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-master release-blocker]
- #89604 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2.0 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*gorm.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.gorm [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6879272&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6879272&tab=artifacts#/gorm) on release-22.1 @ [c4b58a9c755b98e985c1948f42cc10c79fa030e0](https://github.com/cockroachdb/cockroach/commits/c4b58a9c755b98e985c1948f42cc10c79fa030e0):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/gorm/run_1
	orm_helpers.go:193,orm_helpers.go:119,java_helpers.go:220,gorm.go:127,test_runner.go:883: 
		Tests run on Cockroach v22.1.8-123-gc4b58a9c75
		Tests run against gorm v1.24.0
		1 Total Tests Run
		0 tests passed
		1 test failed
		0 tests skipped
		0 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.[build failed] - unknown (unexpected)
		For a full summary look at the gorm artifacts 
		
		An updated blocklist (gormBlocklist22_1) is available in the artifacts' gorm log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89606 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2]
- #89605 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-master]
- #89604 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*gorm.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.gorm [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6897515&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6897515&tab=artifacts#/gorm) on release-22.1 @ [e438c2f89282e607e0e6ca1d38b2e0a622f94493](https://github.com/cockroachdb/cockroach/commits/e438c2f89282e607e0e6ca1d38b2e0a622f94493):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/gorm/run_1
	orm_helpers.go:193,orm_helpers.go:119,java_helpers.go:220,gorm.go:127,test_runner.go:883: 
		Tests run on Cockroach v22.1.8-129-ge438c2f892
		Tests run against gorm v1.24.0
		1 Total Tests Run
		0 tests passed
		1 test failed
		0 tests skipped
		0 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.[build failed] - unknown (unexpected)
		For a full summary look at the gorm artifacts 
		
		An updated blocklist (gormBlocklist22_1) is available in the artifacts' gorm log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89729 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #89606 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2]
- #89605 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-master]
- #89604 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*gorm.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.gorm [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6918242&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6918242&tab=artifacts#/gorm) on release-22.1 @ [4d1f1989e4e13a6e99aff6c08643ab2faed0b744](https://github.com/cockroachdb/cockroach/commits/4d1f1989e4e13a6e99aff6c08643ab2faed0b744):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/gorm/run_1
	orm_helpers.go:193,orm_helpers.go:119,java_helpers.go:220,gorm.go:127,test_runner.go:883: 
		Tests run on Cockroach v22.1.8-137-g4d1f1989e4
		Tests run against gorm v1.24.0
		1 Total Tests Run
		0 tests passed
		1 test failed
		0 tests skipped
		0 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.[build failed] - unknown (unexpected)
		For a full summary look at the gorm artifacts 
		
		An updated blocklist (gormBlocklist22_1) is available in the artifacts' gorm log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89729 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #89606 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2]
- #89605 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-master]
- #89604 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*gorm.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.gorm [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=6937683&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=6937683&tab=artifacts#/gorm) on release-22.1 @ [46963373ba1650d93a77f7b1359f8bf37b928f31](https://github.com/cockroachdb/cockroach/commits/46963373ba1650d93a77f7b1359f8bf37b928f31):


```
The test failed on branch=release-22.1, cloud=gce:
test artifacts and logs in: /artifacts/gorm/run_1
	orm_helpers.go:193,orm_helpers.go:119,java_helpers.go:220,gorm.go:127,test_runner.go:883: 
		Tests run on Cockroach v22.1.8-145-g46963373ba
		Tests run against gorm v1.24.0
		1 Total Tests Run
		0 tests passed
		1 test failed
		0 tests skipped
		0 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.[build failed] - unknown (unexpected)
		For a full summary look at the gorm artifacts 
		
		An updated blocklist (gormBlocklist22_1) is available in the artifacts' gorm log
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #89729 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-21.2]
- #89606 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2]
- #89605 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-master]
- #89604 roachtest: gorm failed [C-test-failure O-roachtest O-robot T-sql-experience branch-release-22.2.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*gorm.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/gorm.go
+++ b/pkg/cmd/roachtest/tests/gorm.go
@@ -24,7 +24,7 @@ import (
 )
 
 var gormReleaseTag = regexp.MustCompile(`^v(?P<major>\d+)\.(?P<minor>\d+)\.(?P<point>\d+)$`)
-var gormSupportedTag = "v1.23.8"
+var gormSupportedTag = "v1.24.0"
 
 func registerGORM(r registry.Registry) {
 	runGORM := func(ctx context.Context, t test.Test, c cluster.Cluster) {
@@ -93,10 +93,8 @@ func registerGORM(r registry.Registry) {
 			t.Fatal(err)
 		}
 
-		blocklistName, expectedFailures, ignorelistName, ignoredFailures := gormBlocklists.getLists(version)
-		if expectedFailures == nil {
-			t.Fatalf("No gorm blocklist defined for cockroach version %s", version)
-		}
+		blocklistName, expectedFailures := "gormBlocklist", gormBlocklist
+		ignorelistName, ignoredFailures := "gormIgnorelist", gormIgnorelist
 		t.L().Printf("Running cockroach version %s, using blocklist %s, using ignorelist %s", version, blocklistName, ignorelistName)
 
 		err = c.RunE(ctx, node, `./cockroach sql -e "CREATE DATABASE gorm" --insecure`)
@@ -113,10 +111,14 @@ func registerGORM(r registry.Registry) {
 		t.Status("running gorm test suite and collecting results")
 
 		// Ignore the error as there will be failing tests.
+		// TODO(rafi): migrate_test.go is removed here since it relies on
+		// multi-dimensional arrays, which aren't supported, and leads to a panic in
+		// the test runner.
 		err = c.RunE(
 			ctx,
 			node,
-			fmt.Sprintf(`cd %s && GORM_DIALECT="postgres" GORM_DSN="user=root password= dbname=gorm host=localhost port=26257 sslmode=disable"
+			fmt.Sprintf(`cd %s && rm migrate_test.go &&
+				GORM_DIALECT="postgres" GORM_DSN="user=root password= dbname=gorm host=localhost port=26257 sslmode=disable"
 				go test -v ./... 2>&1 | %s/bin/go-junit-report > %s`,
 				gormTestPath, goPath, resultsPath),
 		)

--- a/pkg/cmd/roachtest/tests/gorm_blocklist.go
+++ b/pkg/cmd/roachtest/tests/gorm_blocklist.go
@@ -10,30 +10,6 @@
 
 package tests
 
-var gormBlocklists = blocklistsForVersion{
-	{"v20.2", "gormBlocklist20_2", gormBlocklist20_2, "gormIgnorelist20_2", gormIgnorelist20_2},
-	{"v21.1", "gormBlocklist21_1", gormBlocklist21_1, "gormIgnorelist21_1", gormIgnorelist21_1},
-	{"v21.2", "gormBlocklist21_2", gormBlocklist21_2, "gormIgnorelist21_2", gormIgnorelist21_2},
-	{"v22.1", "gormBlocklist22_1", gormBlocklist22_1, "gormIgnorelist22_1", gormIgnorelist22_1},
-	{"v22.2", "gormBlocklist22_2", gormBlocklist22_2, "gormIgnorelist22_2", gormIgnorelist22_2},
-}
+var gormBlocklist = blocklist{}
 
-var gormBlocklist22_2 = gormBlocklist22_1
-
-var gormBlocklist22_1 = gormBlocklist21_2
-
-var gormBlocklist21_2 = gormBlocklist21_1
-
-var gormBlocklist21_1 = gormBlocklist20_2
-
-var gormBlocklist20_2 = blocklist{}
-
-var gormIgnorelist22_2 = gormIgnorelist22_1
-
-var gormIgnorelist22_1 = gormIgnorelist21_2
-
-var gormIgnorelist21_2 = gormIgnorelist21_1
-
-var gormIgnorelist21_1 = gormIgnorelist20_2
-
-var gormIgnorelist20_2 = blocklist{}
+var gormIgnorelist = blocklist{}

